aspnetcore/blazor/blazor-server-ef-core.md

```diff
@@ -0,0 +1,126 @@
+
+### Create new DbContext instances
+
+The fastest way to create a new `DbContext` instance is by using `new` to create a new instance. There are several scenarios that may require additional dependencies to be resolved. For example, you may wish to use [DbContextOptions](https://docs.microsoft.com/ef/core/miscellaneous/configuring-dbcontext#configuring-dbcontextoptions) to configure the context. One approach is to register the options in DI and inject them into the component, like this:
+
+```razor
+@inject DbContextOptions<MyContext> Options
+
+@code {
+    public Task LoadAsync(int id)
+    {
+        using var context = new MyContext(Options);
+        return context.MyEntities.SingleOrDefaultAsync(e => e.Id == id);
+    }
+}
+```
+
+Another solution is to use a factory. EF Core 5.0 (preview) provides a built-in factory for creating new contexts. Because the sample app is based on the latest stable versions and not the preview, it implements its own factory in `Data/DbContextFactory.cs`. 
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Data/DbContextFactory.cs)]
+
+The factory is injected into components and used to create new instances. For example, in `Pages/Index.razor`:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/Index.razor?range=199-212)]
+
+### Scope to the component lifetime
+
+There are two approaches to scope a context to the component lifetime. If the context is registered with DI, you can use the special `OwningComponentBase` base class to create a control scope. Learn more about the class [here](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.components.owningcomponentbase-1).
+
+The second approach is to use the factory to create a context and track it for the lifetime of the component. First, implement `IDisposable` and inject the factory as shown in `Pages/EditContact.razor`.
+
+[!code-razor[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=5-7)]
+
+The sample app ensures the contact is disposed when the component is:
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=181-184)]
+
+Finally, `OnInitializedAsync` is overridden to create a new context. In the sample app, it loads the contact in the same method.
+
+[!code-csharp[](./common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/EditContact.razor?range=89-104)]
+
+## The sample app
+
+The sample application was built as a reference for Blazor Server apps the use EF Core. It includes a grid with sorting and filtering, delete, add, and update operations. It demonstrates use of EF Core to handle optimistic concurrency.
```

Comment:
> It includes a grid 

@SteveSandersonMS Since a bunch of folks will look at this sample, do we need to make sure this grid is aligned with the component authoring performance guidance you've been working on?

Reply:
Since this is inherently Blazor Server-only, it's likely that perf will look good regardless. However in case people copy these patterns for WebAssembly I've posted a couple of suggestions that will help perf.

Reply:
Thanks for the review!